### PR TITLE
[RPD-214] Update documentation for remote state storage ticket

### PR DESCRIPTION
**Pull request by osw282**
This PR updates the documentation on mentioning we are creating new resources (mainly remote state management) and updating Azure Cost diagram as well.

What's new:
- Updated cost diagram.
- `getting-started.md` briefly mentioned remote state manager.
- `inside-matcha.md` explained that there will be a matcha.config.json file generated when provision is finished that will require user to upload to their github repo.
- Fix minor typo in `inside-matcha.md`.

After a discussion, @archena has suggested to break down this ticket such that the [`matcha-under-the-hood`](https://github.com/fuzzylabs/matcha/blob/develop/docs/img/inside-matcha/matcha-under-the-hood.png) diagram will be updated in the next ticket [[RPD-222](https://fuzzy-labs.atlassian.net/jira/software/projects/RPD/boards/4?selectedIssue=RPD-222) Update matcha under the hood diagram].

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [x] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Other (add details above)

[RPD-222]: https://fuzzy-labs.atlassian.net/browse/RPD-222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dudeperf3ct — COMMENTED** [review]
Will there be a different section for `force-unlock` command?

**Comment by osw282** [comment]
> 

Yeah, I am thinking about that too, I original mentioned that the `force-unlock` command should be documented in this PR. What's the status of that PR, we can either create another ticket to document `force-unlock`, doing it in [[RPD-222](https://fuzzy-labs.atlassian.net/jira/software/projects/RPD/boards/4?selectedIssue=RPD-222) Update matcha under the hood diagram] or wait for that ticket to finish and document it in this PR. What do you think?

File: docs/inside-matcha.md [hunk 1]
```diff
@@ -17,11 +17,13 @@ Fundamentally, `provision` stands up the default infrastructure that we've speci
...
+When `provision` is run by the user, we take their input (the `region` and `prefix`) and populate a set of Terraform files - our hand crafted sensible defaults defined as infrastructure-as-code. Once we have the populated Terraform files, Matcha calls `init` (via the [python-terraform](https://github.com/beelit94/python-terraform) library) to download the information we need from Azure which is used for deploying infrastructure. Immediately after, `apply` is run which deploys the infrastructure defined in the Terraform files to Azure.
 
-Once the provisioning on Azure has completed, information about the resources are stored in a `matcha.state` file which, along with the populated Terraform files, are stored in a `.matcha/` directory.
+After completing the provisioning process on Azure, the output information of the remote state manager is stored in a `matcha.config.json` file in the project root directory. Additionally, information about the provisioned resources, along with the populated Terraform files, is stored in a `.matcha/infrastructure` directory in a `matcha.state` file.
```
Comment (Christopher-Norman) [inline docs/inside-matcha.md:22]: NIT: "..is stored in a `matcha.state` file within a `.matcha/infrastructure` directory."

**Review by Christopher-Norman — APPROVED** [review]
LGTM, couple NITS

**Review by dudeperf3ct — APPROVED** [review]

**Pull request merged**